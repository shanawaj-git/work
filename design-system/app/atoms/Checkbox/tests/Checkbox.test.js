import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';
import CheckboxComponentIndex from '../index';

const shallowRenderer = new ShallowRenderer();

describe('Checkbox.js', () => {
  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = renderer.create(
        <CheckboxComponentIndex
          color="primary"
          value="check"
          label="test"
          labelplacement="end"
        />,
      );
      expect(tree).toBeTruthy();
    });

    it('Should display Checkbox with primary color style', () => {
      shallowRenderer.render(
        <CheckboxComponentIndex
          color="primary"
          value="check"
          label="test"
          labelplacement="end"
        />,
      );
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('Should display disabled button', () => {
      shallowRenderer.render(
        <CheckboxComponentIndex
          disabled
          color="primary"
          value="check"
          label="test"
          labelplacement="end"
        />,
      );
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });
  });
});
