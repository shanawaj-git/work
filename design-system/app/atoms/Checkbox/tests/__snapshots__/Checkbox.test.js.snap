// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.js index Should display Checkbox with primary color style 1`] = `
<ForwardRef(Styled(Component))
  color="primary"
  label="test"
  labelplacement="end"
  value="check"
/>
`;

exports[`Checkbox.js index Should display disabled button 1`] = `
<ForwardRef(Styled(Component))
  color="primary"
  disabled={true}
  label="test"
  labelplacement="end"
  value="check"
/>
`;
